Fix RangeError in `not export` error with other file type

Shinyaigeek opened the pull request:
<!---
Thanks for filing a pull request 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting since someone might have pushed the same thing before!
-->

# ↪️ Pull Request

Fixes: #7278

Hi team ! 👋 
I'm interested in contributing to parcel, and I have been doing some research on #7278, and I noticed that this RangeError is caused by language mismatch in highlighting. For example, in the situation of this issue, the JavaScript highlight that is the source of the css import is being done as a css one, and the resulting change in the number of lines is causing a RangeError.

I fixed this Range error by setting the language of codeFrame in `A does not export B` error to the file type of the file from which A imports (JavaScript in the above example). However, I am not yet familiar with the inner code of parcel, and there may be some context that I missed. If you find something I missed, please feel free to point it out to me.

<!---
Provide a general summary of the pull request here
Please look for any issues that this PR resolves and tag them in the PR.
-->

## 💻 Examples

<!-- Examples help us understand the requested feature better -->

```javascript
import React from "react";
import * as styles from "./hoge.module.css";

const App = function() {
  return (
    <div className={styles.notExisting} />
  )
}
```

```css
// empty css
```

## 🚨 Test instructions

<!-- In case it is impossible (or too hard) to reliably test this feature/fix with unit tests, please provide test instructions! -->

I was wondering how to test it, and I wrote it in `scope-hoisting` integration test, but it may not be the right test to write here. If you have a better idea, I'd appreciate it if you could point it out.

For now, I wrote a test to see if the language of the codeFrame is javascript instead of css in the `hoge.module.css does not export notExisting` error in a case like an example above

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

Shinyaigeek committed:
Fix set codeFrames' language from incomingDeps' source type

Shinyaigeek committed:
Chore add test for setting correct language in not export error

Shinyaigeek committed:
Merge branch 'v2' into fix/codeFrames-language

mischnic commented on packages/core/core/src/requests/AssetGraphRequest.js:
```diff
@@ -554,7 +554,9 @@ export class AssetGraphBuilder {
                           loc?.filePath,
                         ) ?? undefined,
-                      language: assetNode.value.type,
+                      language:
+                        incomingDep.value.sourceAssetType ??
+                        assetNode.value.type,
```
`assetNode.value.type` is never correct, so this should rather be
```suggestion
                      language: incomingDep.value.sourceAssetType ?? undefined,
```

Shinyaigeek replied:
OK! Thank you for helping 👍

mischnic reviewed (commented):
Great find! Could you also move the test to `packages/core/integration-tests/test/postcss.js` and call it something like "should throw an error when importing a missing class"?

Here's an example for a test using scope hoisting in that file:
https://github.com/parcel-bundler/parcel/blob/f65182989334638e5fca7d4ce637bdf471b9b874/packages/core/integration-tests/test/postcss.js#L156-L195

Shinyaigeek committed:
Update: not set not export error's  codeFrame language assetNode.value.type

Co-authored-by: Niklas Mischkulnig <4586894+mischnic@users.noreply.github.com>

Shinyaigeek committed:
Chore: mv test for not exporting error into postcss integration test

Shinyaigeek commented:
Thank you for reviewing and helping! I moved the test into `packages/core/integration-tests/test/postcss.js`, and fixed the test label to "should throw an error when importing a missing class”? in f5ca1bc 👍

Shinyaigeek committed:
Merge branch 'v2' into fix/codeFrames-language
